sklearn/metrics/classification.py

```diff
@@ -292,6 +294,154 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None):
     return CM
 
 
+def multilabel_confusion_matrix(y_true, y_pred, sample_weight=None,
+                                labels=None, samplewise=False):
+    """Returns a confusion matrix for each output of a multilabel problem
+
+    Multiclass tasks will be treated as if binarised under a one-vs-rest
+    transformation. Returned confusion matrices will be in the order of
+    sorted unique labels in the union of (y_true, y_pred).
+
+    Parameters
+    ----------
+    y_true : array or sparse, shape (n_samples, n_outputs) or (n_samples,)
+    y_pred : array or sparse, shape (n_samples, n_outputs) or (n_samples,)
+    sample_weight : array, (n_samples,), optional
+    labels : array-like
+        A list of classes or column indices to select some (or to force
+        inclusion of classes absent from the data)
+    samplewise : bool, default=False
+        In the multilabel case, this calculates a confusion matrix per sample
+
+    Returns
+    -------
+    multi_confusion : array, shape (n_outputs, 2, 2)
+        A 2x2 confusion matrix corresponding to each output in the input.
+    """
+    y_true = check_array(y_true, ensure_2d=False, dtype=None,
+                         accept_sparse=['csr', 'csc'])
+    y_pred = check_array(y_pred, ensure_2d=False, dtype=None,
+                         accept_sparse=['csr', 'csc'])
+    check_consistent_length(y_true, y_pred, sample_weight)
+    if sample_weight is not None and sample_weight.ndim > 1:
+        raise ValueError('sample_weight should be 1-d array. ')
+    if sample_weight is not None and samplewise:
+        raise ValueError('sample_weight should not be used with samplewise. ')
+
+    y_type, y_true, y_pred = _check_targets(y_true, y_pred)
```

Comment:
(I wonder if we could do something more lightweight than a full `_check_targets`, seeing as we ultimately do `unique_labels` repetitively, but let's leave that for future work)